### PR TITLE
WEB 541 Fontawesome

### PR DESCRIPTION
**Pull request by dmetzner**
- replaced all images that could have been replaced by fontawesome,
- updated design to fit fontawesome instead of image (the look should be close to the old one)
- added additional design changes to fit better for fontawesome (like buttons, etc)
- fixed minor css issues for themes and removed tables (now css alignment)

Now everything should be in the correct theme colors. (except default avatar image)

- Added icons to to the profile and myprofile page ;)

### CODE REVIEW

File: app/Resources/views/program.html.twig [hunk 1]
```diff
@@ -72,32 +72,31 @@
           <div id="program-like-buttons-container">
             <a id="program-like-thumbs-up" class="program-like-button" href="#"
                data-like-type="{{ constant('Catrobat\\AppBundle\\Entity\\ProgramLike::TYPE_THUMBS_UP') }}">
-              <img class="img-share" src="{{ asset('images/social/like/thumbs_up.png') }}" alt="Thumbs up!"
-                   title="Thumbs up!"/>
+              <div class="like-round-icon-container mr-3"><i class="like-round-icon fas fa-thumbs-up"></i></div>
```
Comment (HCrane) [inline app/Resources/views/program.html.twig]: Please add tooltips as it was before.

File: app/Resources/views/program.html.twig [hunk 2]
```diff
@@ -72,32 +72,31 @@
...
               <span
                   class="singular">{{ "programs.likeThumbsUpDetailText"|transchoice(1, { '%persons%' : 1 }, "catroweb") }}</span>
               <span
                   class="plural">{{ "programs.likeThumbsUpDetailText"|transchoice(2, { '%persons%' : 2 }, "catroweb") }}</span>
             </a>
             <a id="program-like-smile" class="program-like-button" href="#"
                data-like-type="{{ constant('Catrobat\\AppBundle\\Entity\\ProgramLike::TYPE_SMILE') }}">
-              <img class="img-share" src="{{ asset('images/social/like/smile.png') }}" alt="Smile!" title="Smile!"/>
+              <i class="like-icon mr-3 fas fa-grin-squint"></i>
```
Comment (HCrane) [inline app/Resources/views/program.html.twig]: Tooltips

File: app/Resources/views/program.html.twig [hunk 3]
```diff
@@ -72,32 +72,31 @@
...
               <span
                   class="singular">{{ "programs.likeSmileDetailText"|transchoice(1, { '%persons%' : 1 }, "catroweb") }}</span>
               <span
                   class="plural">{{ "programs.likeSmileDetailText"|transchoice(2, { '%persons%' : 2 }, "catroweb") }}</span>
             </a>
             <a id="program-like-love" class="program-like-button" href="#"
                data-like-type="{{ constant('Catrobat\\AppBundle\\Entity\\ProgramLike::TYPE_LOVE') }}">
-              <img class="img-share" src="{{ asset('images/social/like/love.png') }}" alt="Love!" title="Love!"/>
+              <div class="like-round-icon-container mr-3"><i class="like-round-icon-heart fas fa-heart"></i></div>
```
Comment (HCrane) [inline app/Resources/views/program.html.twig]: Tooltips

File: app/Resources/views/program.html.twig [hunk 4]
```diff
@@ -72,32 +72,31 @@
...
               <span
                   class="singular">{{ "programs.likeLoveDetailText"|transchoice(1, { '%persons%' : 1 }, "catroweb") }}</span>
               <span
                   class="plural">{{ "programs.likeLoveDetailText"|transchoice(2, { '%persons%' : 2 }, "catroweb") }}</span>
             </a>
             <a id="program-like-wow" class="program-like-button" href="#"
                data-like-type="{{ constant('Catrobat\\AppBundle\\Entity\\ProgramLike::TYPE_WOW') }}">
-              <img class="img-share" src="{{ asset('images/social/like/wow.png') }}" alt="Wooow!" title="Wooow!"/>
+              <i class="like-icon mr-3 fas fa-surprise"></i>
```
Comment (HCrane) [inline app/Resources/views/program.html.twig]: Tooltips

File: app/Resources/views/program.html.twig [hunk 5]
```diff
@@ -196,98 +195,40 @@
...
           <button id="download" class="btn btn-primary download-button">
-            <table>
-              <tr>
-                <td><img src="{{ asset('images/symbols/download.svg') }}"></td>
-                <td>{{ "download"|trans({}, "catroweb") }}</td>
-              </tr>
-            </table>
+                <img class="program-big-button-icon mr-3"
+                     src="{{ asset('images/symbols/download.svg') }}">
+                <span style="padding-top: 2em; vertical-align: middle">{{ "download"|trans({}, "catroweb") }}</span>
```
Comment (HCrane) [inline app/Resources/views/program.html.twig]: Alignment issues in the button as can be seen here:
image

File: app/Resources/views/program.html.twig [hunk 6]
```diff
@@ -196,98 +195,40 @@
...
         </a>
 
         <button id="remix-graph-button" class="btn btn-primary download-button">
-          <table>
-            <tr>
-              <td><i class="fa fa-sitemap" aria-hidden="true" style="font-size: 40px"></i></td>
-              <td>{{ "remixGraph.showRemixGraph"|trans({}, "catroweb") }}</td>
-            </tr>
-          </table>
+          <i class="program-big-button-icon mr-3 fa fa-sitemap"></i>
```
Comment (HCrane) [inline app/Resources/views/program.html.twig]: It seems the Remixgraph is generating a bug with chrome. We will look at this together.

File: web/font_awesome_wrapper/fontawesome-free/package.json [hunk 7]
```diff
@@ -1,11 +1,11 @@
 {
   "_args": [
     [
-      "@fortawesome/fontawesome-free",
-      "/home/catroweb/Catroweb-Symfony"
+      "@fortawesome/fontawesome-free@^5.3.1",
+      "/home/heroiceraser/Catroweb-Symfony"
```
Comment (HCrane) [inline web/font_awesome_wrapper/fontawesome-free/package.json]: This is suboptimal that it gets changed more often. We need to check if we can cicumvent this.

**Review by HCrane — CHANGES_REQUESTED** [review]
A few changes have to be done but otherwise

| Status | Review Item |
| ------------- | ------------- |
| :heavy_check_mark:  | Codestyle  |
| :heavy_check_mark:   | Code  |
| :heavy_check_mark:  | Tests  |

File: app/Resources/views/program.html.twig (hunk 5 above)
Comment (dmetzner) [reply app/Resources/views/program.html.twig]: + button now only has a min-width

File: web/font_awesome_wrapper/fontawesome-free/package.json (hunk 7 above)
Comment (HCrane) [reply web/font_awesome_wrapper/fontawesome-free/package.json]: We will think about this a little longer.

File: app/Resources/views/program.html.twig (hunk 6 above)
Comment (HCrane) [reply app/Resources/views/program.html.twig]: Was fixed by reloading the site on closing the graph.

**Review by HCrane — APPROVED** [review]

**Pull request merged**